src/app/features/groups/components/group-modal/group-modal.component.ts

```diff
@@ -0,0 +1,111 @@
+import { Component, EventEmitter, Input, OnDestroy, Output, inject, OnInit, effect, computed, Signal } from '@angular/core';
+import { CommonModule } from '@angular/common';
+import { FormGroup, FormsModule, NonNullableFormBuilder, ReactiveFormsModule, Validators } from '@angular/forms';
+
+import { Subject, takeUntil, tap } from 'rxjs';
+
+import { NzModalModule } from 'ng-zorro-antd/modal';
+import { NzFormModule } from 'ng-zorro-antd/form';
+import { NzInputModule } from 'ng-zorro-antd/input';
+import { NzMessageModule, NzMessageService } from 'ng-zorro-antd/message';
+
+import { Group, GroupFormValue } from '../../../../shared/interfaces/groups';
+import { ModalType } from '../../../../shared/interfaces/students';
+import { GroupService } from '../../../../shared/services/group.service';
+
+@Component({
+  selector: 'app-group-modal',
+  standalone: true,
+  imports: [NzModalModule, NzFormModule, NzInputModule, ReactiveFormsModule, FormsModule, CommonModule, NzMessageModule],
```

Comment:
I still recommend you to create a module that would export all of this imports, it will shorten your component definition

Reply:
But Angular 17's standAlone mode doesnt have seperate Module, it is recommended to import inside Component